Added a new use case for transport.port (#42126)
elastic/elasticsearch: docs/reference/modules/transport.asciidoc

within the cluster. Each call that goes from one node to the other uses
the transport module (for example, when an HTTP GET request is processed
by one node, and should actually be processed by another node that holds

> the data). The transport module is also used for other Java Clients. (i.e.

Reviewer: ```suggestion
the data). The transport module is also used for the `TransportClient` in the
{es} Java API.
```

{es} Java API.

> the Cloudian Sync for Object Storage Integration into Elasticsearch)

Reviewer: ```suggestion

```

Reviewer: Sorry. We'll need to remove this as well.

Otherwise, this looks good to me.
